plugins/services/src/js/components/ServiceStatusIcon.js

```diff
@@ -95,28 +97,42 @@ class ServiceStatusWarning extends Component {
   }
 
   render() {
-    const { item } = this.props;
+    const { service } = this.props;
+    const displayName = service.getServiceStatus().displayName;
+    let iconState = displayName;
+
+    if (
+      displayName === ServiceStatusLabels.DEPLOYING ||
+      displayName === ServiceStatusLabels.RECOVERING ||
+      displayName === ServiceStatusLabels.DELETING
+    ) {
+      iconState = "TRANSITION";
+    }
 
-    if (item instanceof ServiceTree) {
-      return this.getServiceTreeWarning(item);
+    iconState = iconState.toUpperCase();
+
+    if (service instanceof ServiceTree) {
+      return (
+        this.getServiceTreeWarning(service) || <StatusIcon state={iconState} />
+      );
     }
 
     // Display the declined offers warning if that's causing a delay. If not,
     // we try to display an unable to launch warning, then default to nothing.
     return (
-      this.getDeclinedOffersWarning(item) ||
-      this.getUnableToLaunchWarning(item) ||
-      <noscript />
+      this.getDeclinedOffersWarning(service) ||
```

Comment:
IMO changing these methods to just return JSX, and moving the conditional logic into the `render` is clearer. Less jumping around tracing things, and the methods will then have a consistent return type, rather than sometimes returning `null`:

```
render() {
 ...
 if (DeclinedOffersUtil.shouldDisplayDeclined()) {
     return this.getDeclinedOffersWarning()
 }

 if (this.shouldShowUnableToLaunch()) {
     return this.getUnableToLaunchWarning()
 }

 return <StatusIcon />
}

```

Reply:
When you don't want to return a component you would do `null` or `<noscript />` as was previously, so I believe returning `null` here is ok.

I also believe the change suggested don't change much.

Reply:
Yes, I agree it's fine either way. It was just a suggestion for what I think is better grouping of the conditionals all in the same place.

Reply:
I have modified to use explicit conditions. thanks